src/hart/hart.cc

```diff
@@ -8,7 +10,67 @@
 
 namespace sim {
 
-Hart::Hart(const fs::path &executable) {
+class InfCache final : public IBBCache {
+  std::unordered_map<Addr, BasicBlock> hash_{};
+
+public:
+  const BasicBlock &
+  lookupUpdate(Addr key, std::function<BasicBlock(Addr)> slowGetData) override {
+    if (auto hit = hash_.find(key); hit != hash_.end())
+      return hit->second;
+
+    auto bb = slowGetData(key);
+    return hash_[key] = std::move(bb);
+  }
```

Comment:
Why don't you use `try_emplace()` here in order to reduce the amount of key searching & insertion
If key is not found, you can move bb to the returned iterator